お問い合わせフォーム最終課題の実装

Kubo-Tomoya opened the pull request:
***お問い合わせフォーム最終課題の実装***

**【実装内容】**
・管理者用のお問い合わせフォーム画面の作成
・管理者が顧客からのお問い合わせを、編集、削除、データベースへの更新処理などの機能を実装

**【主な追加・変更】**
①ContactController
・GET /admin/contacts/{id}/edit: 該当IDのお問い合わせデータを取得し、編集画面に表示
・POST /admin/contacts/{id}/edit: フォーム送信されたデータで該当レコードを更新
・POST /admin/contacts/{id}/delete: 該当データを削除（物理削除）

②contact_edit.html
・お問い合わせ内容（姓、名、メールアドレス、電話番号、住所、種別、本文）を編集可能なフォームを用意
・更新ボタン・削除ボタンを配置し、編集後は /admin/contacts にリダイレクト
・フッター（ログアウトボタン）を共通テンプレート fragments/footer.html として分離・読み込み

③Security Config.java
・CSRFを一時的に無効化（開発段階のため）
・/admin/** 配下のアクセス制限を引き続き適用

Kubo-Tomoya committed:
お問い合わせフォーム最終課題の実装

uchiyama4929 reviewed (changes requested):
全体的にインデントに使っている空白文字列がタブと半角スペースが混在して使用されている状況です。
プロジェクト全体で1種類に統一するのが基本なので下記の記事を参考にまずは可視化設定を行ってください。
https://itsakura.com/eclipse_displaytab

コーティングを行う時に統一するように心がけていきましょう。

uchiyama4929 commented on practice/src/main/resources/templates/admin_signup.html:
```diff
@@ -0,0 +1,22 @@
+		<label>姓:<input type="text" th:field="*{lastName}"></label><br/>
+		<label>名:<input type="text"th:field="*{firstName}"></label><br/>
+		<label>メール:<input type="text" th:field="*{email}"></label><br/>
+		<label>パスワード:<input type="text" th:field="*{password}"></label><br/>
```
emailにするとスマホで表示した時に英字のキーボードがデフォルトで表示されたりします。
telにするとスマホがテンキーを表示したりします。
passwordだと入力した文字列が*****という感じで隠すことができます。

この辺は色々と種類があって基本的にはtextでも問題なく動くのですが微妙に違うので適切なものを選ぶようにしましょう。

またGoogleがサイトを読み込んだ時にタグの意味を理解できるようになるとサイトの理解度がUPし、検索上位に表示されやすくなるみたいな効果もあるので、inputのtypeもそうですが、pタグやdivタグ、sectionタグなどのhtmlタグに関しても少しずつ適切に使用できるように心がけてみてください！

uchiyama4929 replied:
type=emailに設定するとブラウザが勝手にバリデーションをつけるような動きになるのでそのへんはformタグにnovalidate属性をつけると無効にできるのでいらない場合は消しても大丈夫です。

uchiyama4929 replied:
LGTM！

Kubo-Tomoya committed:
4/21　修正依頼の内容の実装

①タブと半角スペースの統一
②admin_signup.htmlのメールtype→email、パスワードをtype→passwordに変更
③ContactControllerにバリデーションを追加
④デバッグ用のコメントを削除

Kubo-Tomoya committed:
4/21　修正依頼の内容の実装

①タブと半角スペースの統一
②admin_signup.htmlの

Kubo-Tomoya committed:
System.out.printlnを削除し、修正

Kubo-Tomoya committed:
コンフリクト解消：PostMapping出力のマージ

uchiyama4929 commented on practice/src/main/java/com/example/demo/controller/ContactController.java:
```diff
@@ -88,4 +86,64 @@ public String complete(Model model, HttpServletRequest request) {
+        if (contactOpt.isPresent()) {
+            Contact contact = contactOpt.get();
+            contact.setLastName(form.getLastName());
+            contact.setFirstName(form.getFirstName());
```
LGTM！

uchiyama4929 commented on practice/src/main/java/com/example/demo/controller/AdminController.java:
```diff
@@ -0,0 +1,62 @@
+    
+    @PostMapping("/signup")
+    public String processSignup(@ModelAttribute Admin admin) {
+    	System.out.println("これできんとおわですわ");
```
LGTM！

Kubo-Tomoya committed:
mainブランチとの競合解消（ContactService.java）

Kubo-Tomoya committed:
ContactServiceImp.javaを削除
